From 896499837132fb9a3c5399112a44a73bef7e6666 Mon Sep 17 00:00:00 2001
From: yifan.bai <yifan.bai@spreadtrum.com>
Date: Fri, 15 Apr 2016 15:25:03 +0800
Subject: [PATCH] Fix ApnSettings crash issue.

Steps:
 1. Add a new APN;
 2. Click 'Reset to default';
 3. During reset, click the new APN;
 4. Crash happened.
Solution:
Disable menu items when reseting default APN.

Change-Id: I8df439d066e58039e9f91a9be642127eb7298159
---

diff --git a/src/com/android/settings/ApnSettings.java b/src/com/android/settings/ApnSettings.java
index fdc0914..21ce123 100644
--- a/src/com/android/settings/ApnSettings.java
+++ b/src/com/android/settings/ApnSettings.java
@@ -358,6 +358,19 @@ public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
     }
 
     @Override
+    public void onPrepareOptionsMenu(Menu menu) {
+        MenuItem item = null;
+        item = menu.findItem(MENU_NEW);
+        if (item != null) {
+            item.setEnabled(!mRestoreDefaultApnMode);
+        }
+        item = menu.findItem(MENU_RESTORE);
+        if (item != null) {
+            item.setEnabled(!mRestoreDefaultApnMode);
+        }
+    }
+
+    @Override
     public boolean onOptionsItemSelected(MenuItem item) {
         switch (item.getItemId()) {
         case MENU_NEW:
@@ -427,6 +440,7 @@ private String getSelectedApnKey() {
     private boolean restoreDefaultApn() {
         showDialog(DIALOG_RESTORE_DEFAULTAPN);
         mRestoreDefaultApnMode = true;
+        getPreferenceScreen().setEnabled(false);
 
         if (mRestoreApnUiHandler == null) {
             mRestoreApnUiHandler = new RestoreApnUiHandler();
@@ -465,6 +479,7 @@ public void handleMessage(Message msg) {
                         getResources().getString(
                                 R.string.restore_default_apn_completed),
                         Toast.LENGTH_LONG).show();
+                    getActivity().invalidateOptionsMenu();
                     break;
             }
         }
